framework/Models/Code.py

```diff
@@ -312,7 +312,7 @@ def createNewInput(self,currentInput,samplerType,**kwargs):
     if not found:
       self.raiseAnError(IOError,'None of the input files has one of the extensions requested by code '
                                   + self.subType +': ' + ' '.join(self.code.getInputExtension()))
-    subDirectory = os.path.join(self.workingDir, kwargs['prefix'] if 'prefix' in kwargs.keys() else '1')
+    subDirectory = os.path.join(self.workingDir, kwargs['prefix'][0] if 'prefix' in kwargs.keys() else '1')
```

Comment:
the "prefix" is always a numpy array? or in other models it is not?

Reply:
If we're consistent in our construction of realizations, it should always be a numpy array, right?

Reply:
You are right, sampler should not be setting this to a numpy array.